static/app/views/discover/savedQuery/utils.tsx

```diff
@@ -198,8 +199,16 @@ export function handleUpdateHomepageQuery(
       addSuccessMessage(t('Saved as Discover default'));
       return savedQuery;
     })
-    .catch(() => {
-      addErrorMessage(t('Unable to set query as Discover default'));
+    .catch((e: RequestError) => {
+      let errorMessage = t('Unable to set query as Discover default');
+
+      if ('responseJSON' in e) {
+        const response = e.responseJSON;
+        if (response?.non_field_errors && Array.isArray(response.non_field_errors)) {
```

Comment:
This is very special cased right now, but we can expand it out for other errors. I just know this is what the error for the `is:` filter looks like.